sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/interface.scala

```diff
@@ -198,6 +200,10 @@ case class CatalogTable(
       locationUri, inputFormat, outputFormat, serde, compressed, properties))
   }
 
+  def withStats(cboStatsEnabled: Boolean): CatalogTable = {
```

Comment:
I can think of two approaches:

1.

We can keep the current naive version of `statistics` and add new `statistics` function which takes conf.

A default implementation of the new `statistics` function simply returns the naive version of `statistics`.

In `Join` or `Aggregate`, we can include more complex logic in the new `statistics` to return naive calculation or something estimation.

The caller always calls new `statistics` function and passes in current conf.

2.

Add new `statisticsCBO` which doesn't take conf because it is called only cbo is enabled. So the caller decides to call non-cbo version `statistics` or cbo version `statisticsCBO`.